core/src/main/java/be/cytomine/controller/appengine/RestTaskController.java

```diff
@@ -25,21 +29,31 @@
 @ConditionalOnExpression("${application.appEngine.enabled: false}")
 public class RestTaskController extends RestCytomineController {
 
-    @Autowired
-    private AppEngineService appEngineService;
+    private final AppEngineService appEngineService;
+
+    private final RestTemplate restTemplate;
 
     @GetMapping("/tasks/{id}")
-    public String descriptionById(
-            @PathVariable String id
-    ) {
+    public String descriptionById(@PathVariable String id) {
         return appEngineService.get("tasks/" + id);
     }
 
     @GetMapping("/tasks/{namespace}/{version}")
     public String description(
             @PathVariable String namespace,
-            @PathVariable String version
+            @PathVariable String version,
+            @RequestParam(required = false) String host
     ) {
+        log.info("GET /tasks/{}/{}?host={}", namespace, version, host);
+        if (host != null) {
+            String url = UriComponentsBuilder
+                .fromUriString(UriUtils.decode(host, StandardCharsets.UTF_8))
+                .pathSegment("api", "v1", "tasks", namespace, version)
+                .toUriString();
+
+            return restTemplate.exchange(url, HttpMethod.GET, null, String.class).getBody();
+        }
```

Comment:
What do you think of that ? 
```suggestion
        @RequestParam(required = false, name = "host") Optional<String> maybeHost
    ) {
        log.info("GET /tasks/{}/{}?host={}", namespace, version, maybeHost);
        return maybeHost
            .map(host -> UriComponentsBuilder
                .fromUriString(UriUtils.decode(host, StandardCharsets.UTF_8))
                .pathSegment("api", "v1", "tasks", namespace, version)
                .toUriString())
            .map(url ->
                restTemplate.exchange(url, HttpMethod.GET, null, String.class).getBody())
            .orElseGet(() -> appEngineService.get("tasks/" + namespace + "/" + version));
    }
```

Reply:
Yes, I think this is better indeed since more Java best practice.